### PR TITLE
cannon: Set notify to true for periodic cannon jobs

### PR DESCRIPTION
**Pull request by mbaxter**
**Description**

Set notify to true for periodic cannon jobs.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1512,6 +1512,8 @@ workflows:
           name: cannon-go-lint-and-test-<<matrix.mips_word_size>>-bit
           requires:
             - contracts-bedrock-build
+          skip_slow_tests: false
+          notify: true
```
Comment (Inphi) [inline .circleci/config.yml:1516]: Could you also add `mentions: "@proofs-team"`?

Comment (mbaxter) [reply .circleci/config.yml:1516]: This is just to be extra explicit?  Looks like it defaults to proofs-team [here](https://github.com/ethereum-optimism/optimism/blob/971eb82eecb9d158788e463d15fbc8d42df7388e/.circleci/config.yml#L245-L249).

Comment (Inphi) [reply .circleci/config.yml:1516]: Nah. I didn't realize it already defaults to proofs. No need then.

**Review by Inphi — APPROVED** [review]

**Pull request merged**